controllers/nutanixcluster_controller.go

```diff
@@ -178,12 +182,21 @@ func (r *NutanixClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	}
 	conditions.MarkTrue(cluster, infrav1.CredentialRefSecretOwnerSetCondition)
 
-	v3Client, err := CreateNutanixClient(ctx, r.SecretInformer, r.ConfigMapInformer, cluster)
+	v3Client, err := prismclientcache.DefaultCache.Get(cluster.Name)
 	if err != nil {
-		conditions.MarkFalse(cluster, infrav1.PrismCentralClientCondition, infrav1.PrismCentralClientInitializationFailed, capiv1.ConditionSeverityError, err.Error())
-		return ctrl.Result{Requeue: true}, fmt.Errorf("nutanix client error: %v", err)
+		if errors.Is(err, prismclientcache.ErrorClientNotFound) {
+			log.Info("Nutanix client not found in cache. Creating new client")
```

Comment:
nit: Nutanix client not found in cache. => Nutanix pc client is not found in cache.